community/bolt/src/docs/dev/serialization.asciidoc

```diff
@@ -420,6 +422,46 @@ Path (signature=0x50) {
 }
 ----
 
+The nodes (`N`) and relationships (`R`) are defined as follows:
+
+- `N` contains all the unique nodes in the path
+- `R` contains all the unique relationships in the path
+- For `N`, the index is an integer commencing with 0 and incrementing by 1
+- For `R`, the index is an integer commencing with 1 and incrementing by 1
+- The value component for both `N` and `R` is the data corresponding to the node or relationship (as defined above)
+- In `N`, the first element must _always_ be the first node in the path (thus having 0 as the key)
+- No other _explicit_ rules apply as to either (i) the ordering of the other nodes in `N`, or (ii) the ordering of any of the relationships in `R`. However, while not required, it is recommended that implementations aim to list entities (i.e. nodes and relationships) in the order in which they are first encountered while traversing the path. This may help with the efficiency of reading and writing
+
+In the context of a transmission, the path is represented as a sequence of integers; we define `S` as the transmitted sequence, and `S'` as the full sequence.
+
+- `S` must always consist of an even number of integers, or be empty
+- The first, third, ... integer in `S` has a range encompassed by (..,-1] and [1,..). These represent the relationships in the path
+- The second, fourth, ... integer in `S` has a range encompassed by [0,..). These represent the nodes in the path
+- By definition, the first node in the path will _always_ have an index of 0, so we exclude this from `S` upon transmission. The idea is to construct `S'` by prepending 0 to `S` on the completion of a successful transmission
+- Let a path `P` be given by the following transmitted sequence [1, 1, -2, 2]. It follows that the corresponding full sequence, `S'`, is given by [0, 1, 1, -2, 2]
+- The first integer in `S` (1) is the index in `R` corresponding to the first relationship in `P`
+- The second integer in `S` (1) is the index in `N` corresponding to the second node in `P`
+- The last integer in `S` (2) is the index in `N` corresponding to the last node in `P`
+- When a relationship is represented by a positive integer in `S` - such as the 1 in position 1 - this means that the relationship is being traversed in the direction of the underlying relationship in the data graph
+- When a relationship is represented by a negative integer in `S` - such as the -2 in position 3 - this means that the relationship is being traversed against the direction of the underlying relationship in the data graph.
+For 'loops' - i.e. a relationship beginning and ending at the same node - a positive integer should be used
+
+===== Example
+
+Consider the following path:
+
+`(A)-[:X]\->(B)-[:Y]\->\(C)\<-[:Z]-(B)\<-[:X]-(A)`
+
+The elements trasmitted would be as follows:
+
+- `N`: `(A)`, `(B)`, `\(C)`
+- `R`: `[:X]`, `[:Y]`, `[:Z]`
+- `S`: [1, 1, 2, 2, -3, 1, -1, 0]
+
+By definition, the following is also implied:
+
+- `S'`: [0, 1, 1, 2, 2, -3, 1, -1, 0]
+
```

Comment:
What about the corner cases... such as zero-length paths? Also, showing examples where a node/s is/are traversed more than once? This is just to preempt questions from people about this...